### ISSUE TITLE
Issue in Mounting

### ISSUE THREAD

**Issue by seeranjeeviramavel** [ghiscoding/slickgrid-react#301]
### Describe the bug

**Issue Description:**
I'm encountering a runtime error while using the slickgrid-react library. The error message indicates "Cannot read properties of null (reading 'id')", and the stack trace points to a specific location within the slickgrid-react codebase.

**Expected Behavior:**
The component should render without any runtime errors, and the slickgrid-react library should function as expected.

**Additional Context:**
Version of slickgrid-react being used.
Relevant code snippets or configurations that might be related to the issue.
Any other relevant information about the environment or setup where the issue is occurring.

**Error Message:**
Cannot read properties of null (reading 'id')

**Here is the copy of the codesanbox i've worked:**
[https://codesandbox.io/p/sandbox/slick-grid-test3-xfs9f8?file=%2Fsrc%2FGridTest%2FGrid.tsx%3A901%2C34-901%2C39](url)

### Expectation

- When installing `slickgrid-react,` it's often necessary to also install several external dependencies. 
- Simplifying this to a single dependency would make the process easier. 
- Additionally, providing a working demo with code on platforms like CodeSandbox or JSFiddle would enhance usability.

### Environment Info

```shell
I'm using codesandbox
```

### Validations

- [X] Follow our [Code of Conduct](https://github.com/ghiscoding/slickgrid-react/blob/master/.github/CODE_OF_CONDUCT.md)
- [X] Read the [HOWTO - Step by Step](https://github.com/ghiscoding/slickgrid-react/wiki/HOWTO--Step-by-Step).
- [X] Check that there isn't [already an issue](https://github.com/ghiscoding/slickgrid-react/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/ghiscoding/slickgrid-react/discussions).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by Pratosh22** [comment]
I had faced the same problem too, You can disable/remove React.strictMode and it should work fine.

**Comment by ghiscoding** [comment]
I'm not a React developer myself, if you know how to resolve this kind of problem then please contribute or else I close this issue since an alternative was provided. Thanks

**Comment by seeranjeeviramavel** [comment]
> I had faced the same problem too, You can disable/remove React.strictMode and it should work fine.

Thanks. It works <3

**Comment by ghiscoding** [comment]
I took a quick look at that React StrictMode and from what I can see, it's way too strict and doesn't make any sense in some cases since it goes way further than what TypeScript already does. It even fails in SortableJS which is an external library that I have no control of. I added couple of extra checks but that doesn't seem to be enough to cover everything because like I said, it's way too strict and I don't think it will be possible to ever support that strict mode... so with that in mind, I'm closing this issue, just go disable strict mode like suggested. Thanks
